File: leetCode/Easy/0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        map.put(target-nums[0] ,0);

        for(int i=1; i<nums.length; i++) {
            if(map.containsKey(nums[i])) {
                return new int[]{ map.get(nums[i]), i};
            } else {
                map.put(target-nums[i], i);
            }
        }
        return new int[]{};
        
    }
}

File: leetCode/Easy/0001-two-sum/0001-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf(target-nums[0] to 0)

        for(i in 1..nums.lastIndex) {
            if(map.containsKey(nums[i])) {
                return intArrayOf(map[nums[i]]!!, i)
            } else {
                map[target-nums[i]] = i
            }
        }
        return intArrayOf()
    }
}